components/profile-controller/pkg/controller/profile/profile_controller.go

```diff
@@ -119,346 +120,163 @@ func (r *ReconcileProfile) Reconcile(request reconcile.Request) (reconcile.Resul
-					"secrets",
-					"serviceaccounts",
-					"services",
-					"services/proxy",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"",
-				},
-				Resources: []string{
-					"bindings",
-					"events",
-					"limitranges",
-					"pods/log",
-					"pods/status",
-					"replicationcontrollers/status",
-					"resourcequotas",
-					"resourcequotas/status",
-				},
-				Verbs: []string{
-					"get",
-					"list",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"",
-				},
-				Resources: []string{
-					"serviceaccounts",
-				},
-				Verbs: []string{
-					"impersonate",
-				},
-			},
-			{
-				APIGroups: []string{
-					"apps",
-				},
-				Resources: []string{
-					"daemonsets",
-					"deployments",
-					"deployments/rollback",
-					"deployments/scale",
-					"replicasets",
-					"replicasets/scale",
-					"statefulsets",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"autoscaling",
-				},
-				Resources: []string{
-					"horizontalpodautoscalers",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"batch",
-				},
-				Resources: []string{
-					"cronjobs",
-					"jobs",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"extensions",
-				},
-				Resources: []string{
-					"daemonsets",
-					"deployments",
-					"deployments/rollback",
-					"deployments/scale",
-					"ingresses",
-					"networkpolicies",
-					"replicasets",
-					"replicasets/scale",
-					"replicationcontrollers/scale",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
-			{
-				APIGroups: []string{
-					"policy",
-				},
-				Resources: []string{
-					"poddistruptionbudgets",
-				},
-				Verbs: []string{
-					"create",
-					"delete",
-					"get",
-					"list",
-					"patch",
-					"update",
-					"watch",
-				},
-			},
+	}
+	if err := controllerutil.SetControllerReference(profileIns, serviceAccount, r.scheme); err != nil {
+		return err
+	}
+	found := &corev1.ServiceAccount{}
+	err := r.Get(context.TODO(), types.NamespacedName{Name: serviceAccount.Name, Namespace: serviceAccount.Namespace}, found)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating ServiceAccount", "namespace", serviceAccount.Namespace,
+				"name", serviceAccount.Name)
+			err = r.Create(context.TODO(), serviceAccount)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
```

Comment:
I guess you don't need to update service account as it doesn't have any spec?

Reply:
Yes